fdbclient/NativeAPI.actor.cpp

```diff
@@ -2742,6 +2743,33 @@ ACTOR static Future<Void> tryCommit( Database cx, Reference<TransactionLogInfo>
 						trLogInfo->addLog(FdbClientLogEvents::EventCommit(startTime, latency, req.transaction.mutations.size(), req.transaction.mutations.expectedSize(), req));
 					return Void();
 				} else {
+					// clear the RYW transaction which contains previous conflicting keys
+					tr->info.conflictingKeysRYW.reset();
+					if (ci.conflictingKRIndices.present()){
+						// In general, if we want to use getRange to expose conflicting keys, we need to support all the parameters getRange provides.
+						// It is difficult to take care of all corner cases of what getRange does.
+						// Consequently, we use a hack way here to achieve it.
+						// We create an empty RYWTransaction and write all conflicting key/values to it.
+						// Since it is RYWTr, we can call getRange on it with same parameters given to the original getRange.
+						tr->info.conflictingKeysRYW = std::make_shared<ReadYourWritesTransaction>(tr->getDatabase());
+						// To make the getRange call local, we need to explicitly set the read version here.
+						// This version number 100 set here does nothing but prevent getting read version from the proxy
+						tr->info.conflictingKeysRYW->setVersion(100);
+						// Clear the whole key space, thus, RYWTr knows to only read keys locally
+						tr->info.conflictingKeysRYW->clear(normalKeys);
+						// in case system keys are conflicting
+						tr->info.conflictingKeysRYW->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+						tr->info.conflictingKeysRYW->clear(systemKeys);
+						// merge duplicate indices
+						const auto cKRs = ci.conflictingKRIndices.get();
+						std::set<int> mergedIds(cKRs.begin(), cKRs.end());
+						for (auto const & rCRIndex : mergedIds) {
```

Comment:
This implementation is relying on the read conflict ranges not overlapping. This is true when the read conflicts come from ReadYourWrites transactions, but not necessarily true if someone is using NativeAPI directly or has turned on READ_YOUR_WRITES_DISABLE. You should keep track of the largest end key you have seen, and if the beginning of a range is less than that, do a much more complicated merge (I think you should be able to use `krmSetRange`).

Reply:
you should modify your test so it can detect this bug

Reply:
Done.
The transactions in the test workload are randomly enabled `READ_YOUR_WRITES_DISABLE ` option.
Since `READ_YOUR_WRITES_DISABLE` does the same as native `transaction`'s `addReadConflictRange` and `addWriteConflictRange`, we only need to test one of them here.